rs/messaging/src/routing/stream_builder.rs

```diff
@@ -399,6 +408,25 @@ impl StreamBuilderImpl {
                             oversized_requests.push(req);
                         }
 
+                        // Best-effort request to unsupported subnet. Always route subnet-local requests
+                        // for consistency with scheduler routing.
+                        //
+                        // TODO(MR-649): Drop this once best-effort calls are fully deployed.
+                        RequestOrResponse::Request(req)
+                            if dst_subnet_id != self.subnet_id
```

Comment:
AFAICT this is the only potentially contentious decision.

I could have gone the other way (also prevent routing in the scheduler), but that would have required even more code and more tests. Or, we can go with Message Routing rejecting local best-effort calls and the scheduler routing them, but that seems  confusing and potentially problematic.